controlplane/kubeadm/internal/control_plane.go

```diff
@@ -253,6 +253,11 @@ func (c *ControlPlane) HasUnhealthyMachine() bool {
 	return len(c.UnhealthyMachines()) > 0
 }
 
+// HasHealthyMachineStillProvisioning returns true if any healthy machine in the control plane is still in the process of being provisioned.
+func (c *ControlPlane) HasHealthyMachineStillProvisioning() bool {
+	return len(c.Machines.Filter(collections.Not(collections.HasUnhealthyCondition), collections.And(collections.Not(collections.HasNode())))) > 0
```

Comment:
```suggestion
	return len(c.Machines.Filter(collections.Not(collections.HasUnhealthyCondition), collections.Not(collections.HasNode()))) > 0
```

I think the And is not doing anything (at least with the current code)

Reply:
Improved the filter and added unit tests